src/domain/prompt/BasePrompt.ts

```diff
@@ -86,9 +89,25 @@ export abstract class BasePrompt<PromptConfigType extends IPromptConfig<PromptCo
   }
 
   /** Whether or not a value has been set on this instance. */
-  get isEmpty() {
+  get isEmpty(): boolean {
     return this.value === undefined
   }
+  
+  get block(): IBlock | undefined {
+    const ctx = this.runner.context
+    const intx = findInteractionWith(this.interactionId, ctx)
+    const flow = findFlowWith(intx.flowId, ctx)
+
+    try {
+      return findBlockWith(intx.blockId, flow)
+    } catch (e) {
+      if (!(e instanceof ValidationException)) {
```

Comment:
It's an exceptionally picky thing, but this should probably invert the `!(e instanceof ValidationException)` so it is a positive, and then return if it is an instance of ValidationException, otherwise, rethrowing e, as that is easier to read, and if a new error type needs to be handled here, it would avoid a potential mistake of this if statement catching it instead.

Reply:
Ah, I always struggle with this one, because I try to make the happy path the _intent_. Whereas the guards are the unhappy paths through the code. 

For me, this makes things more readable in that skimming a method's top level code will cover the happy path. Then I can circle on a second pass to see which scenarios have been guarded against. 

What are your thoughts on this?

Reply:
I think it's different, as this is a catch block, and not the complete body of a function. If it was the complete body of a function, I would totally agree, because, you want to fail fast out of those. However, in a catch block, you are already in the middle (or usually the end) of a function body, and because it is part-way through, it is much easier to miss. I have gotten caught by this exact situation before, and not realized that a catch block in a multi-hundred line function had a statement like this, halfway through the huge catch handler, and it was very easy to miss a simple `!` on a statement.